### PR TITLE
LLD - address param added to accounts deeplink

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

address param added to accounts deeplink

### ❓ Context

- **Impacted projects**: `ledger-live-desktop` 
- **Linked resource(s)**: [LIVE-5869] 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-5869]: https://ledgerhq.atlassian.net/browse/LIVE-5869?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/hooks/useDeeplinking.js [hunk 1]
```diff
@@ -87,6 +87,16 @@ export function useDeepLinkHandler() {
...
       switch (url) {
         case "accounts":
+          const { address } = query;
+          if (address && typeof address === "string") {
+            const account = accounts.find(acc => acc.id === address)
+            if (account.type === "Account") {
+              navigate(`/account/${account.id}`);
+            } else {
+              navigate(`/account/${account.parentId}/${account.id}`);
+            }
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-desktop/src/renderer/hooks/useDeeplinking.js:97]: What happen if no account is found?

**Review by gre — CHANGES_REQUESTED** [review]
do we have a way to test this? @elGrogz was asking for help on this & i think we really need to have tests on these (looking at the number of times we had incident around deeplink)

File: apps/ledger-live-desktop/src/renderer/hooks/useDeeplinking.js [hunk 2]
```diff
@@ -87,6 +87,18 @@ export function useDeepLinkHandler() {
...
+            const account = accounts.find(acc => acc.freshAddress === address)
+            if (account) {
+              if (account.type === "Account") {
+                navigate(`/account/${account.id}`);
+              } else {
+                navigate(`/account/${account.parentId}/${account.id}`);
+              }
+              break;
+            }
+          }
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/hooks/useDeeplinking.js:101]: please rebase on develop which probably will be an auto merge. useDeeplinking.js moved to useDeeplinking.ts

https://github.com/LedgerHQ/ledger-live/pull/3066

File: apps/ledger-live-desktop/src/renderer/hooks/useDeeplinking.ts [hunk 3]
```diff
@@ -82,6 +82,18 @@ export function useDeepLinkHandler() {
...
       switch (url) {
         case "accounts":
+          const { address } = query;
+          if (address && typeof address === "string") {
+            const account = accounts.find(acc => acc.freshAddress === address)
+            if (account) {
+              if (account.type === "Account") {
+                navigate(`/account/${account.id}`);
+              } else {
+                navigate(`/account/${account.parentId}/${account.id}`);
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/hooks/useDeeplinking.ts:92]: this case is not possible to happen. accounts is an array of Account. on the other example it's based on bellow, getAccountsOrSubAccountsByCurrency was in used and it made more sense in that case.

Comment (cgrellard-ledger) [reply apps/ledger-live-desktop/src/renderer/hooks/useDeeplinking.ts]: Thanks Gaetan, it's fixed ! 👍

**Review by mcayuelas-ledger — APPROVED** [review]

**Pull request merged**